### ISSUE TITLE
Revise NumericDecimals regex to handle scientific notation?

### ISSUE THREAD

**Issue by gutibeni** [Mikuana/rumydata#109]
https://github.com/Mikuana/rumydata/blob/dbac94cd6801c7b76e4fe04e03776407819016ca/rumydata/rules/cell.py#L298

Some of the files I'm validating contain values in Excel's scientific notation (e.g, -1.2234E-05). Although I've set the precision to 38 digits, the rumydata is raising an error on these values -- I believe due to the non-numeric digit (E). I haven't fully tested it out, but revising the NumericDecimals regex to this might work: ` -?\d+(\.\d{1,' + str(self.decimals) + '})?([eE]-?\d+)?  `  

It's possible that this revision would introduce unwanted errors. I thought it was still worth bringing attention to the scientific notation issue. Thanks!

**Comment by Mikuana** [comment]
@gutibeni I've pushed up a change which adds a new type of field called `Number`. It has a parameter which lets you control whether scientific notation is accepted. You would use it like the example below.

```python
from rumydata import Layout
from rumydata.field import Text, Integer, Choice, Number

layout = Layout(definition={
    'col1': Text(8),
    'col2': Choice(['x', 'y', 'z']),
    'col3': Integer(1)
    'col4': Number(9, allow_scientific=True)
})
```

Try installing the code from this branch and see if it works for your use case.

```shell
 pip install "rumydata @ git+https://github.com/Mikuana/rumydata.git@scientific-notation"
```

**Comment by gutibeni** [comment]
Thanks for adding the `Number` field! It's a great solution, and it works perfectly for my use case.

**Comment by Mikuana** [comment]
Implemented in release v1.5.0
